src/test/kotlin/study/StringCalculator.kt

```diff
@@ -0,0 +1,90 @@
+package study
+
+import java.util.LinkedList
+import java.util.Queue
+import java.util.function.BiFunction
+
+class StringCalculator(private val splitter: String = " ") {
+    fun calculate(source: String?): Double {
+        val value = validate(source)
+        val numberStack = value.toNumberStack(splitter)
+        val operatorQueue = value.toOperatorQueue(splitter)
+        while (!(numberStack.size == 1 && operatorQueue.isEmpty())) {
+            val result = calculate(numberStack, operatorQueue)
+            numberStack.push(result)
+        }
+        return numberStack.peek().toDouble()
+    }
+
+    private fun validate(source: String?): String {
+        if (source.isNullOrBlank()) {
+            throw IllegalArgumentException("빈 값이나 공백이 들어올 수 없습니다.")
+        }
+        return source
+    }
+
+    private fun calculate(numbers: LinkedList<Double>, operators: Queue<Operator>): Double {
+        if (numbers.size < 2 || operators.isEmpty()) {
+            throw IllegalArgumentException("잘못된 수식입니다.")
+        }
```

Comment:
```suggestion
        require(numbers.size > 2 || operators.isNotEmpty()) { throw IllegalArgumentException("잘못된 수식입니다.") }
```

kotlin에서는 `require` 함수를 통해서 `IllegalArgumentException`을 처리할 수 있습니다! ☺️

참고

- [Kotlin 예외처리](https://seosh817.tistory.com/155)